app/views/layouts/_announcement.html.erb

```diff
@@ -0,0 +1,8 @@
+<% if announcement_visible?(current_announcement) %>
+  <div id="announcement">
+    <span><%= current_announcement.body.html_safe %></span>
+    <span class="hide">
+      <%= link_to "hide", "#", onclick: "hideAnnouncement('#{current_announcement.token}')" %>
```

Comment:
Please don't use inline javascript, you would end up violating [our CSP](https://github.com/rubygems/rubygems.org/blob/master/app/controllers/application_controller.rb#L15). Perhaps you can add it to `announcement,js`.